raphtory/src/python/packages/algorithms.rs

```diff
@@ -145,6 +146,22 @@ pub fn in_components(g: &PyGraphView) -> AlgorithmResult<DynamicGraph, Vec<GID>,
     components::in_components(&g.graph, None)
 }
 
+/// In component -- Finding the "in-component" of a node in a directed graph involves identifying all nodes that can be reached following only incoming edges.
+///
+/// Arguments:
+///     node (Node) : The node whose in-component we wish to calculate
+///
+/// Returns:
+///    An array containing the Nodes within the given nodes in-component
+#[pyfunction]
+#[pyo3(signature = (node))]
+pub fn in_component(node: &PyNode) -> Vec<PyNode> {
```

Comment:
just return `Vec<NodeView<DynamicGraph>>` here and let pyo3 figure out the rest (it's probably more efficient as it can directly construct a python list instead of making another vector first.